mitmproxy/proxy/layers/http/_http2.py

```diff
@@ -120,6 +120,7 @@ def is_open_for_us(self, stream_id: int) -> bool:
     def _handle_event(self, event: Event) -> CommandGenerator[None]:
         if isinstance(event, Start):
             self.h2_conn.initiate_connection()
+            self.h2_conn.increment_flow_control_window(2**24)
```

Comment:
Two thoughts:

1. I don't understand why this is necessary on top of the settings. If it's indeed required, let's add a short comment explaining why.
2. Flow control is entirely handled in `_http_h2.py`. So should we maybe override `initiate_connection` connection there so that everything flow-control related is at one place?